elm/wizard.py

```diff
@@ -234,7 +246,15 @@ def chat(self, query,
             response_message += ref_msg
             if stream:
                 print(ref_msg)
-
+        end_time = perf_counter()
+        total_chat_time = start_chat_time - end_time
+        performance = {
+            "total_chat_time": total_chat_time,
+            "chat_completion_time": chat_completion_time,
```

Comment:
am i understanding correctly that the chat completion time is time to get first response from the LLM? That's great and important but if the LLM is in "stream" mode, does this actually work or does the API return something immediately but it only starts responding in the `for chunk in response` generator loop? If this is the case, maybe we should have the end time be calculated at the first chunk generation in the response object.